index.js

```diff
@@ -67,10 +64,10 @@ S3Storage.prototype._handleFile = function (req, file, cb) {
         if (ev.total) currentSize = ev.total
       })
 
-      upload.send(function (err) {
+      upload.send(function (err, data) {
         if (err) return cb(err)
 
-        cb(null, { size: currentSize, key: filePath })
+        cb(null, extend(data, { size: currentSize, key: key }))
```

Comment:
It contains `Location` and `Etag`. Location is extremely useful for ensuring that the file was uploaded to the expected place and provides an absolute URL to the asset. Etag is used to ensure proper cache control of assets to prevent re-fetching if the binary has not changed.